### ISSUE TITLE
Unable to play geoblocked content

### ISSUE THREAD

**Issue by WimDeWinter01** [add-ons/plugin.video.vrt.nu#888]
### Describe the bug ###

When I want to watch geo blocked content I get an error and am requested to add a Belgian cell phone number to my profile. My profile has a Belgian phone number and address. I can watch this content on my smartphone using the vrtnu app. How can I remedy this?

### To Reproduce ###
1. Go to addons > VRTNU
2. Click on any geoblocked content
3. See error

### Expected behavior ###

The content playing without error

### Additional context ###

- Operating system: Manjaro linux
- Kodi version: 19.1.1 (latest in the repos)
- Addon version: 2.5.1+matrix.1
- Using a VPN: yes, azireVPN on both manjaro (using openVPN) and smartphone (using the app, and that's wireguard based)
- Country you are using the addon from: Belgium, vpn server in Amsterdam

### Log (if available) ###
```paste below

2021-05-25 22:17:58.264 T:2376    ERROR <general>: Control 55 in window 10025 has been asked to focus, but it can't
2021-05-25 22:17:58.278 T:2376    DEBUG <general>: CGUIMediaWindow::GetDirectory (plugin://plugin.video.vrt.nu/programs/babylon-berlin/3)
2021-05-25 22:17:58.278 T:2376    DEBUG <general>:   ParentPath = [plugin://plugin.video.vrt.nu/programs/babylon-berlin/3]
2021-05-25 22:17:58.279 T:7873    DEBUG <general>: Thread waiting start, auto delete: false
2021-05-25 22:17:58.286 T:7868    DEBUG <general>: CAddonDatabase::SetLastUsed[plugin.video.vrt.nu] took 7 ms
2021-05-25 22:17:58.286 T:7873    DEBUG <general>: StartScript - calling plugin VRT NU('plugin://plugin.video.vrt.nu/programs/babylon-berlin/3','1','','resume:false')
2021-05-25 22:17:58.286 T:7873    DEBUG <general>: GetLanguageInvoker - Reusing LanguageInvokerThread 3 for script /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py
2021-05-25 22:17:58.286 T:7874    DEBUG <general>: Thread JobWorker start, auto delete: true
2021-05-25 22:17:58.287 T:3713     INFO <general>: initializing python engine.
2021-05-25 22:17:58.287 T:3713    DEBUG <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): start processing
2021-05-25 22:17:58.287 T:3713    DEBUG <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): entering source directory /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib
2021-05-25 22:17:58.287 T:3713    DEBUG <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): instantiating addon using automatically obtained id of "plugin.video.vrt.nu" dependent on version 3.0.0 of the xbmc.python api
2021-05-25 22:17:58.289 T:3713    DEBUG <CAddonSettings[plugin.video.vrt.nu]>: trying to load setting definitions from old format...
2021-05-25 22:17:58.291 T:3713     INFO <general>: [plugin.video.vrt.nu] Access: plugin://plugin.video.vrt.nu/programs/babylon-berlin/3
2021-05-25 22:17:58.291 T:3713    DEBUG <general>: [plugin.video.vrt.nu][routing] Dispatching to 'programs', args: {'program': 'babylon-berlin', 'season': '3'}
2021-05-25 22:17:58.292 T:3713  WARNING <general>: [plugin.video.vrt.nu] Got item from cache '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/cache/favorites.json'
2021-05-25 22:17:58.295 T:3713  WARNING <general>: [plugin.video.vrt.nu] Got item from cache '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/cache/resume_points.json'
2021-05-25 22:17:58.296 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL get: https://vrtnu-api.vrt.be/search?i=video&size=300&facets[allowedRegion]=[BE,WORLD]&facets[programUrl]=//www.vrt.be/vrtnu/a-z/babylon-berlin/&facets[seasonTitle]=3
2021-05-25 22:17:58.379 T:2376    DEBUG <general>: ------ Window Init (DialogBusy.xml) ------
2021-05-25 22:17:58.480 T:2376    DEBUG <general>: CGLContextEGL::SwapBuffers: sync check blocking
2021-05-25 22:17:58.495 T:2376    DEBUG <general>: CGLContextEGL::SwapBuffers: sync sleep: 14748
2021-05-25 22:17:58.582 T:3713     INFO <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): script successfully run
2021-05-25 22:17:58.583 T:7873    DEBUG <general>: Thread waiting 140367290680896 terminating
2021-05-25 22:17:58.595 T:2376    DEBUG <general>: ------ Window Deinit (DialogBusy.xml) ------
2021-05-25 22:17:58.598 T:7875    DEBUG <general>: Thread BackgroundLoader start, auto delete: false
2021-05-25 22:17:58.601 T:7874    DEBUG <general>: ffmpeg[0x7fa9500ea010X]: [image2] Custom AVIOContext makes no sense and will be ignored with AVFMT_NOFILE format.
2021-05-25 22:17:58.601 T:7876    DEBUG <general>: Thread JobWorker start, auto delete: true
2021-05-25 22:17:58.601 T:7868    DEBUG <general>: ffmpeg[0x55e8816a2080X]: [image2] Custom AVIOContext makes no sense and will be ignored with AVFMT_NOFILE format.
2021-05-25 22:17:58.604 T:7875    DEBUG <general>: Thread BackgroundLoader 140367477872192 terminating
2021-05-25 22:17:59.382 T:2376    DEBUG <general>: CGLContextEGL::SwapBuffers: sync check blocking
2021-05-25 22:17:59.396 T:2376    DEBUG <general>: CGLContextEGL::SwapBuffers: sync sleep: 14051
2021-05-25 22:17:59.646 T:2376    DEBUG <general>: Keyboard: scancode: 0x24, sym: 0x13, unicode: 0x000d, modifier: 0x0
2021-05-25 22:17:59.813 T:2376    DEBUG <general>: Keyboard: scancode: 0x24, sym: 0x13, unicode: 0x0000, modifier: 0x0
2021-05-25 22:17:59.813 T:2376    DEBUG <general>: HandleKey: return (0xf00d) pressed, action is Select
2021-05-25 22:17:59.815 T:2376    DEBUG <general>: OnPlayMedia plugin://plugin.video.vrt.nu/play/id/vid-030c9e96-e714-4247-9030-7ce80f1bd833/pbs-pub-fc1e2388-3ceb-49ce-9c43-531b36c21a1c
2021-05-25 22:17:59.816 T:2376    DEBUG <general>: StartScript - calling plugin VRT NU('plugin://plugin.video.vrt.nu/play/id/vid-030c9e96-e714-4247-9030-7ce80f1bd833/pbs-pub-fc1e2388-3ceb-49ce-9c43-531b36c21a1c','1','','resume:false')
2021-05-25 22:17:59.816 T:2376    DEBUG <general>: GetLanguageInvoker - Reusing LanguageInvokerThread 3 for script /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py
2021-05-25 22:17:59.816 T:3713     INFO <general>: initializing python engine.
2021-05-25 22:17:59.816 T:3713    DEBUG <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): start processing
2021-05-25 22:17:59.816 T:3713    DEBUG <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): entering source directory /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib
2021-05-25 22:17:59.817 T:3713    DEBUG <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): instantiating addon using automatically obtained id of "plugin.video.vrt.nu" dependent on version 3.0.0 of the xbmc.python api
2021-05-25 22:17:59.818 T:3713    DEBUG <CAddonSettings[plugin.video.vrt.nu]>: trying to load setting definitions from old format...
2021-05-25 22:17:59.820 T:3713     INFO <general>: [plugin.video.vrt.nu] Access: plugin://plugin.video.vrt.nu/play/id/vid-030c9e96-e714-4247-9030-7ce80f1bd833/pbs-pub-fc1e2388-3ceb-49ce-9c43-531b36c21a1c
2021-05-25 22:17:59.821 T:3713    DEBUG <general>: [plugin.video.vrt.nu][routing] Dispatching to 'play_id', args: {'video_id': 'vid-030c9e96-e714-4247-9030-7ce80f1bd833', 'publication_id': 'pbs-pub-fc1e2388-3ceb-49ce-9c43-531b36c21a1c'}
2021-05-25 22:17:59.822 T:3713  WARNING <general>: [plugin.video.vrt.nu] Got item from cache '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/tokens/user_XVRTToken.tkn'
2021-05-25 22:17:59.823 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL post: https://media-services-public.vrt.be/vualto-video-aggregator-web/rest/external/v1/tokens
2021-05-25 22:17:59.823 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL post data: b'{"identityToken": "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwczovL3Rva2VuLnZydC5iZSIsInN1YiI6Il9ndWlkXzl1MDNKcUp1TzFfWENmR0RLTzZwcndhbEZqR095NFNycUpleDZ5MTF3Njg9IiwibmV3c3ViIjoiYWM2MTk2NmEtZjBlNy00MTUzLWFmZTgtODZlMjJhZDkyNjAzIiwiY291IjoiQkUiLCJhZ2UiOjQ0LCJhdWQiOiIvdnJ0bnUiLCJpYXQiOjE2MjE5NzIxNDksImV4cCI6MTYyMTk3NTc0OSwiYWMiOiIxNisiLCJ1c2VyX3N0YXR1cyI6IlVOS05PV04ifQ.cwkMF-3R2opp6xJ5UB0gKUYgz4vwEkJYzbf_2whM3d0uTjqP9UEtkWjCYrw7CcKoB7C2_aMbD_2qySuSwFRC3thOlPQ4u557BickP9d4kIv1BFDoZoYtWWMr4plPZvVgeVzufG-PJRNqDAsN5sKdNGI3JW-7K3uEUD1Q_EFaI9FoSiIqR4x4haTYULlyfQGos95nHRH7_ES56kXfDo9l5UigttCfW6DWzqr5dGGd18f7EwOpeeA6BrjXTaVYCf2HfpTnpWwaEg5Wq6Ro779mdL1o4A2ZZRUQ2CHL4G4DHStUnOydIUBCcwy4zRnkewNjWnIVqKlZ1Q0sFanTGZdCgx38ChBzbjVvIMjpcBAOpIQ_pigK9gJGG35cCZ4M9OvJEdbTFgNE9YNqiwgGDn761L7-L3scI_cU3o24OG4fBt3_ZG-4L9YbUDb7L2Qb4MiZzLIccDO3x6DMtI8Of5RiidG0PW4xMS0KJL9TCeBxT4NgweolhV-m_YBH0koU0qlfBCdDkKNQl-PXTtZSax8iDrtN_XC95W3x2CSepfM_soQd0viipnr84X8eGTSjj45tZ4j4bVTzsMybhd0EFw1ZFXJYn8fYh3b7ZaclhO51g3AyQNNH7XM7RCbm6C1-2s5aLqM8r98B1M8x_845QxuhT5wL6fmCADuNZOZpuRLhfaU"}'
2021-05-25 22:17:59.836 T:7879    DEBUG <general>: Thread scriptobs start, auto delete: false
2021-05-25 22:17:59.989 T:3713    DEBUG <general>: [plugin.video.vrt.nu] Write cache '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/tokens/ondemand_vrtPlayerToken.tkn'.
2021-05-25 22:17:59.990 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL get: https://media-services-public.vrt.be/vualto-video-aggregator-web/rest/external/v1/videos/pbs-pub-fc1e2388-3ceb-49ce-9c43-531b36c21a1c$vid-030c9e96-e714-4247-9030-7ce80f1bd833?vrtPlayerToken=e1016@cee9ca799f7746991564f0cfdd165fd17d765f155f7796b67386cfd45ff9e1ae&client=vrtvideo@PROD
2021-05-25 22:18:00.036 T:2376    DEBUG <general>: ------ Window Init (DialogBusy.xml) ------
2021-05-25 22:18:00.132 T:2376    DEBUG <general>: CGLContextEGL::SwapBuffers: sync check blocking
2021-05-25 22:18:00.147 T:2376    DEBUG <general>: CGLContextEGL::SwapBuffers: sync sleep: 14838
2021-05-25 22:18:00.189 T:3713    FATAL <general>: [plugin.video.vrt.nu] VRT Geoblock: Roaming settings are incomplete, to enable watching geo-blocked content while roaming please add additional information to the user profile.
2021-05-25 22:18:00.190 T:3713    DEBUG <general>: [plugin.video.vrt.nu] Delete file '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/tokens/ondemand_vrtPlayerToken.tkn'.
2021-05-25 22:18:00.191 T:3713  WARNING <general>: [plugin.video.vrt.nu] Got item from cache '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/tokens/vrtloginrt.tkn'
2021-05-25 22:18:00.192 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL get: https://token.vrt.be/refreshtoken?legacy=true
2021-05-25 22:18:00.453 T:3713    DEBUG <general>: [plugin.video.vrt.nu] Write cache '/home/wim/.kodi/userdata/addon_data/plugin.video.vrt.nu/tokens/vrtloginat.tkn'.
2021-05-25 22:18:00.455 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL get: https://token.vrt.be/vrtnuinitloginEU?destination=https://www.vrt.be/vrtnu/
2021-05-25 22:18:00.607 T:3713  WARNING <general>: [plugin.video.vrt.nu] URL get: https://login.vrt.be/authorize?access_type=offline&response_type=code&state=05673083d6aef95f26ee925b18bb4fc88bebbd1a5eae74b0b9212f043641568a.aHR0cHM6Ly93d3cudnJ0LmJlL3ZydG51Lw==&client_id=vrtnu-site&scope=openid legacyid email profile mid address video phone&redirect_uri=https://token.vrt.be/vrtnucallback&id_token_hint=eyJraWQiOiJyc2ExIiwiYWxnIjoiUlMyNTYifQ.eyJhdWQiOiJ2cnRudS1zaXRlIiwic3ViIjoiYWM2MTk2NmEtZjBlNy00MTUzLWFmZTgtODZlMjJhZDkyNjAzIiwiaXNzIjoiaHR0cHM6XC9cL2xvZ2luLnZydC5iZSIsInNjb3BlcyI6ImFkZHJlc3Msb3BlbmlkLHByb2ZpbGUsbGVnYWN5aWQsbWlkLGVtYWlsIiwiZXhwIjoxNjIxOTc3NDg5LCJpYXQiOjE2MjE5NzM4ODksImp0aSI6ImM1Mjg0ZmZmLTA5NjQtNGE5Mi05NzZhLTZjOWI0OTY4YjUwNCJ9.Khs0wQjOchwtqra99TDLBxywoL3zkarvsaJDXIPfqA6-D3GpB6yhZtkS9b28vSvLkA8g34PY0wKpkGqMH05k56nShQES3Q4C4dm_WJx_uGkOoqcrOROcELkHE1mgu-CIOhu2aPbeq9Q5FRV2gA9Z25ZEXdVrlk5V9jqa16FQFYE
2021-05-25 22:18:00.848 T:2376    DEBUG <general>: ------ Window Init (DialogConfirm.xml) ------
2021-05-25 22:18:07.052 T:2376    DEBUG <general>: Keyboard: scancode: 0x24, sym: 0x13, unicode: 0x000d, modifier: 0x0
2021-05-25 22:18:07.219 T:2376    DEBUG <general>: Keyboard: scancode: 0x24, sym: 0x13, unicode: 0x0000, modifier: 0x0
2021-05-25 22:18:07.219 T:2376    DEBUG <general>: HandleKey: return (0xf00d) pressed, action is Select
2021-05-25 22:18:07.455 T:2376    DEBUG <general>: ------ Window Deinit (DialogConfirm.xml) ------
2021-05-25 22:18:07.470 T:3713     INFO <general>: CPythonInvoker(3, /home/wim/.kodi/addons/plugin.video.vrt.nu/resources/lib/addon_entry.py): script successfully run
2021-05-25 22:18:07.486 T:2376    DEBUG <general>: ------ Window Deinit (DialogBusy.xml) ------
2021-05-25 22:18:07.486 T:7879    DEBUG <general>: Thread scriptobs 140367248717376 terminating
2021-05-25 22:18:07.487 T:2376    ERROR <general>: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.vrt.nu/play/id/vid-030c9e96-e714-4247-9030-7ce80f1bd833/pbs-pub-fc1e2388-3ceb-49ce-9c43-531b36c21a1c]
2021-05-25 22:18:07.487 T:2376    DEBUG <general>: Playlist Player: no more playable items... aborting playback
```

**Comment by WimDeWinter01** [comment]
I tried starting playing the geoblocked content on the Manjaro machine with the vpn turned off. Then, when it was playing, I turned the vpn on. There were no errors and I am now able to play all geo-blocked content.

**Comment by WimDeWinter01** [comment]
By closing and restarting the addon the issue returns. The "solution" mentioned above isn't persistent.

**Comment by mediaminister** [comment]
I currently can't debug this because my VPN is blacklisted as gets a PROXY_DETECTED error.

**Comment by mediaminister** [comment]
@WimDeWinter01 Can you try if it works on the VRT NU website: https://www.vrt.be/vrtnu/a-z/babylon-berlin/3/babylon-berlin-s3a8/

**Comment by mediaminister** [comment]
Okay, I got a working VPN, but I can't reproduce this issue. Geoblocked content is working.

But, you need VRT NU Add-on **v2.5.3** and it's recommended to remove VRT-tokens in the VRT NU Add-on Expert settings

**Comment by WimDeWinter01** [comment]
Ok, seems the addon works, but my phone number had yet to be validated. This I noticed when I tried to watch using the VRT NU website. A pop-up asked me to validate my phone number. This I did and I was able to play geo-blocked content on the website and add-on. This pop-up did not show up on the add-on.

I also reinstalled the add-on and it keeps working.

Big thanks to mediaminister for helping me out. Thanks!
